draft-westerlund-tsvwg-dtls-over-sctp-bis.md

```diff
@@ -508,40 +508,93 @@ ULP:  Upper Layer Protocol
+   way.
+   After sending the KeyUpdate message, it stops sending DTLS messages until
+   the corresponding Ack message has been processed.
+
+   Prior to processing a received KeyUpdate message, all other received SCTP
+   user messages that are buffered in the SCTP layer and can be delivered to
+   the DTLS layer MUST be read and processed by DTLS.
+
 ## DTLS Epochs
 
-   In general, DTLS implementations SHOULD discard records from
-   earlier epochs, as described in Section 4.2.1 of
-   {{I-D.ietf-tls-dtls13}}. To avoid discarding messages, the
-   processing guidelines in Section 4.2.1 of DTLS 1.3
-   {{I-D.ietf-tls-dtls13}} or Section 4.1 or DTLS 1.2 {{RFC6347}}
-   should be followed.
+   In general, DTLS implementations SHOULD discard records from earlier epochs.
+   However, in the context of a reliable communication this is not appropriate.
+
+### DTLS 1.2 Considerations
+
+   The procedures of Section 4.1 of {{RFC6347}} MUST NOT be followed.
+   Instead, when currently using epoch n, for n > 1, DTLS packets from epoch
+   n - 1 and n MUST be processed.
+
+### DTLS 1.3 Considerations
+
+   The procedures of Section 4.2.1 of {{RFC6347}} are irrelevant.
+   When receiving DTLS packets using epoch n, no DTLS packets from earlier
+   epochs are received.
 
 ## Handling of Endpoint-Pair Shared Secrets
 
    SCTP-AUTH {{RFC4895}} is keyed using Endpoint-Pair Shared
    Secrets. In SCTP associations where DTLS is used, DTLS is used to
    establish these secrets. The endpoints MUST NOT use another
    mechanism for establishing shared secrets for SCTP-AUTH.
-
    The endpoint-pair shared secret for Shared Key Identifier 0 is
-   empty and MUST be used when establishing a DTLS connection.  In
-   DTLS 1.2, whenever the main secret changes, a 64-byte shared secret
-   is derived from every main secret and provided as a new
-   endpoint-pair shared secret by using the TLS-Exporter. In DTLS 1.3,
-   the exporter_secret never change. For DTLS 1.3, the exporter is
-   described in {{RFC8446}}. For DTLS 1.2, the exporter is described
-   in {{RFC5705}}. The exporter MUST use the label given in Section
-   {{IANA-Consideration}} and no context.  The new Shared Key
-   Identifier MUST be the old Shared Key Identifier incremented by 1.
-   If the old one is 65535, the new one MUST be 1.
-
-   Before sending the DTLS Finished message, the active SCTP-AUTH key
+   empty and MUST be used when establishing a DTLS connection.
+
+### DTLS 1.2 Considerations
+
+   Whenever the master secret changes, a 64-byte shared secret is derived from
+   every master secret and provided as a new endpoint-pair shared secret by
+   using the TLS-Exporter described in {{RFC5705}}.
+   The 64-byte shared secret MUST be provided to the SCTP stack as soon as
+   the computation is possible.
+   The exporter MUST use the label given in Section {{IANA-Consideration}}
+   and no context.
+   The new Shared Key Identifier MUST be the old Shared Key Identifier
+   incremented by 1.
```

Comment:
As far as I know the epoch can't wrap in DTLS 1.2 and DTLS 1.3 anymore. Since the epoch and the key id are both 16 bit, the key id can't wrap anymore. In DTLS 1.0 the epoch could wrap, therefore the key id.